worktree_status.go

```diff
@@ -159,3 +163,103 @@ func diffTreeIsEquals(a, b noder.Hasher) bool {
 
 	return bytes.Equal(hashA, hashB)
 }
+
+// Add adds the file contents of a file in the worktree to the index.
+func (w *Worktree) Add(path string) (plumbing.Hash, error) {
+	s, err := w.Status()
+	if err != nil {
+		return plumbing.ZeroHash, err
+	}
+
+	if s.File(path).Worktree == Unmodified {
```

Comment:
I wonder if returning a zero hash when the operation is successful but unmodified is the right behavior. If it is, please, explain it in godoc, it is not that intuitive.